src/main/java/io/cdap/plugin/dlp/configs/DlpFieldTransformationConfig.java

```diff
@@ -0,0 +1,173 @@
+
+      fieldTransformationBuilder.setInfoTypeTransformations(
+        InfoTypeTransformations.newBuilder().addTransformations(infoTypeTransformationBuilder));
+    }
+
+    return fieldTransformationBuilder.build();
+
+  }
+
+
+  DlpFieldTransformationConfig(String transform, String[] fields, String[] filters,
+    DlpTransformConfig transformProperties) {
+    this.transform = transform;
+    this.fields = fields;
+    this.filters = filters;
+    this.transformProperties = transformProperties;
+  }
+
+  public void validate(FailureCollector collector, Schema inputSchema, final String widgetName) {
+    // No need to validate 'transform' field since it is used to deserialize this object
+    // So any invalid values would have been caused an error during deserialization
+
+    Gson gson = new Gson();
+    ErrorConfig errorConfig = getErrorConfig();
+    if (fields.length == 0) {
+      errorConfig.setTransformPropertyId("fields");
+      collector.addFailure(String.format("No fields were selected to apply '%s' transform.", this.transform), "")
+               .withConfigElement(widgetName, gson.toJson(errorConfig));
+    }
+
+    List<Schema.Type> supportedTypes = transformProperties.getSupportedTypes();
+
+    for (String fieldName : this.fields) {
+      Schema.Field field = inputSchema.getField(fieldName);
+      if (field == null) {
+        errorConfig.setTransformPropertyId("fields");
+        collector.addFailure(String.format("Field '%s' is not present in the input schema", fieldName), "")
+                 .withConfigElement(widgetName, gson.toJson(errorConfig));
+      } else {
+
+        Schema.Type fieldType =
+          field.getSchema().isNullable() ? field.getSchema().getNonNullable().getType() : field.getSchema().getType();
+        if (!supportedTypes.contains(fieldType)) {
+          errorConfig.setTransformPropertyId("fields");
+          collector
+            .addFailure(String.format("Field '%s' has type '%s' which is not supported by '%s' transform", fieldName,
+                                      fieldType.toString(), this.transform), "")
+            .withConfigElement(widgetName, gson.toJson(errorConfig));
+        }
+
+      }
+
+
+    }
+
+    if (filters.length == 0) {
+      errorConfig.setTransformPropertyId("filters");
+      collector.addFailure("At least one filter must be selected.", "")
+               .withConfigElement(widgetName, gson.toJson(errorConfig));
+    }
+
+    transformProperties.validate(collector, widgetName, getErrorConfig());
+  }
+
+  public String getTransform() {
+    return transform;
+  }
+
+  public String[] getFields() {
+    return fields;
+  }
+
+  public String[] getFilters() {
+    return filters;
+  }
+
+  public DlpTransformConfig getTransformProperties() {
+    return transformProperties;
+  }
+
+  public ErrorConfig getErrorConfig() {
+    return getErrorConfig(null);
+  }
+
+  public io.cdap.plugin.dlp.configs.ErrorConfig getErrorConfig(String transformPropertyId) {
```

Comment:
why fully qualified classname is needed here?

Reply:
They're not needed, fixed it